lib/spack/spack/bootstrap/core.py

```diff
@@ -228,7 +228,9 @@ def _install_and_test(
                 if not abstract_spec.intersects(candidate_spec):
                     continue
 
-                if python_spec is not None and python_spec not in abstract_spec:
+                if python_spec is not None and not abstract_spec.intersects(
+                    spack.spec.Spec(f"^{python_spec}")
```

Comment:
This should autoconvert str -> Spec, right? saves two lines

Reply:
Yep. It turns out we have a few type hints that are too restrictive. I just fixed the ones in `Spec` here.